### PR TITLE
Memory Leak Fix

### PR DESCRIPTION
**Pull request by brandon1024**
## Fixes https://github.com/brandon1024/node-skeleton/issues/19

Built a session store that uses knex to serialize session data to the database instead of using the default `MemoryStore`. This also allows sessions to persist when the application restarts.

I also added some migration run configurations.

Note: you will need to run migrations.

@jacsmith21 can you have a look at this and see if you can catch any bugs or anything I missed?

### CODE REVIEW

File: .idea/runConfigurations/Run_Migrations.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="Run Migrations" type="js.build_tools.npm" factoryName="npm">
+    <package-json value="$PROJECT_DIR$/package.json" />
+    <command value="run" />
+    <scripts>
+      <script value="migrate" />
```
Comment (jsmith) [inline .idea/runConfigurations/Run_Migrations.xml:6]: How exactly does this ```<script value="migrate" />``` work?

Comment (jsmith) [reply .idea/runConfigurations/Run_Migrations.xml:6]: Nvm I though migrate was a file. I see it now!

**Comment by jsmith** [comment]
Hm I'm getting an error when I run the migrations:
```
Knex:warning - migration failed with error: create table `sessions` (`id` int unsigned not null auto_increment primary key, `sid` varchar(255), `session` varchar(255), `created_at` timestamp default CURRENT_TIMESTAMP, `updated_at` timestamp default CURRENT_TIMESTAMP) default character set utf8 - ER_TABLE_EXISTS_ERROR: Table 'sessions' already exists
Error: ER_TABLE_EXISTS_ERROR: Table 'sessions' already exists
    at Query.Sequence._packetToError (/home/eigen/Desktop/node-skeleton/node_modules/mysql/lib/protocol/sequences/Sequence.js:52:14)
    at Query.ErrorPacket (/home/eigen/Desktop/node-skeleton/node_modules/mysql/lib/protocol/sequences/Query.js:77:18)
    at Protocol._parsePacket (/home/eigen/Desktop/node-skeleton/node_modules/mysql/lib/protocol/Protocol.js:279:23)
    at Parser.write (/home/eigen/Desktop/node-skeleton/node_modules/mysql/lib/protocol/Parser.js:76:12)
    at Protocol.write (/home/eigen/Desktop/node-skeleton/node_modules/mysql/lib/protocol/Protocol.js:39:16)
    at Socket.<anonymous> (/home/eigen/Desktop/node-skeleton/node_modules/mysql/lib/Connection.js:103:28)
    at Socket.emit (events.js:160:13)
    at addChunk (_stream_readable.js:269:12)
    at readableAddChunk (_stream_readable.js:256:11)
    at Socket.Readable.push (_stream_readable.js:213:10)
    at TCP.onread (net.js:599:20)
    --------------------
    at Protocol._enqueue (/home/eigen/Desktop/node-skeleton/node_modules/mysql/lib/protocol/Protocol.js:145:48)
    at Connection.query (/home/eigen/Desktop/node-skeleton/node_modules/mysql/lib/Connection.js:208:25)
    at /home/eigen/Desktop/node-skeleton/node_modules/knex/lib/dialects/mysql/index.js:157:18
    at Promise._execute (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/debuggability.js:303:9)
    at Promise._resolveFromExecutor (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/promise.js:483:18)
    at new Promise (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/promise.js:79:10)
    at Client_MySQL._query (/home/eigen/Desktop/node-skeleton/node_modules/knex/lib/dialects/mysql/index.js:151:12)
    at Client_MySQL.query (/home/eigen/Desktop/node-skeleton/node_modules/knex/lib/client.js:205:17)
    at /home/eigen/Desktop/node-skeleton/node_modules/knex/lib/transaction.js:274:21
    at tryCatcher (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/util.js:16:23)
    at Function.Promise.attempt.Promise.try (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/method.js:39:29)
    at Client_MySQL.trxClient.query (/home/eigen/Desktop/node-skeleton/node_modules/knex/lib/transaction.js:271:34)
    at Runner.<anonymous> (/home/eigen/Desktop/node-skeleton/node_modules/knex/lib/runner.js:149:36)
    at Runner.tryCatcher (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/util.js:16:23)
    at Runner.query (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/method.js:15:34)
    at Runner.<anonymous> (/home/eigen/Desktop/node-skeleton/node_modules/knex/lib/runner.js:212:19)
    at Runner.tryCatcher (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/util.js:16:23)
    at Object.gotValue (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/reduce.js:157:18)
    at Object.gotAccum (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/reduce.js:144:25)
    at Object.tryCatcher (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/eigen/Desktop/node-skeleton/node_modules/bluebird/js/release/promise.js:569:18)
```

I had to drop the sessions table before I was able to run the migration. What table did I just drop :question: 

Now I'm getting this error when I try to access the login page:
```
_http_outgoing.js:500
    throw new errors.Error('ERR_HTTP_HEADERS_SENT', 'set');
    ^

Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at validateHeader (_http_outgoing.js:500:11)
    at ServerResponse.setHeader (_http_outgoing.js:507:3)
    at ServerResponse.header (/home/eigen/Desktop/node-skeleton/node_modules/express/lib/response.js:730:10)
    at ServerResponse.contentType (/home/eigen/Desktop/node-skeleton/node_modules/express/lib/response.js:558:15)
    at ServerResponse.send (/home/eigen/Desktop/node-skeleton/node_modules/express/lib/response.js:145:14)
    at done (/home/eigen/Desktop/node-skeleton/node_modules/express/lib/response.js:967:10)
    at load (/home/eigen/Desktop/node-skeleton/node_modules/twig/twig.js:7177:18)
    at loadTemplateFn (/home/eigen/Desktop/node-skeleton/node_modules/twig/twig.js:5507:18)
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:528:3)
```

File: .idea/runConfigurations/Run_Migrations.xml (hunk 1 above)
Comment (brandon1024) [reply .idea/runConfigurations/Run_Migrations.xml:6]: :+1:

**Comment by brandon1024** [comment]
Ahh ok, so when the SessionStore constructor is called, if the `sessions` table doesn't exist it is created right in the constructor. So if you ran the app before running the migration, the table would have been created and then the migration would break.

One way we could handle this is to just throw an error in the constructor if the table exists with a helpful message `No sessions table exist. Consider running migrations`?

**Comment by brandon1024** [comment]
Regarding the second error, I’ll take a look as soon as I get home tonight and I’ll let you know :) I'm assuming your up to date on this branch? Try doing a complete rollback of your dev database and then rerun migrations.

**Comment by jsmith** [comment]
Sure I'll try on my desktop which hasn't run the migrations yet!

**Pull request merged**